Sources/BraintreeVenmo/BTVenmoAppSwitchRequestURL.swift

```diff
@@ -0,0 +1,104 @@
+import Foundation
+
+#if canImport(BraintreeCore)
+import BraintreeCore
+#endif
+
+// TODO: Entire class be internal and likely a struct once rest of Venmo is in Swift
+// TODO: methods can all be non-static likely when we convert the rest of Venmo to Swift
+@objcMembers public class BTVenmoAppSwitchRequestURL: NSObject {
+
+    static let kXCallbackTemplate: String = "scheme://x-callback-url/path"
+    static let kVenmoScheme: String = "com.venmo.touch.v2"
+
+    /// The base app switch URL for Venmo. Does not include specific parameters.
+    // TODO: property can be internal once rest of Venmo is in Swift
+    public static var baseAppSwitchURL: URL? {
+        appSwitchBaseURLComponents()?.url
+    }
+
+    /// Create an app switch URL
+    /// - Parameters:
+    ///   - merchantID: The merchant ID
+    ///   - accessToken: The access token used by the Venmo app to tokenize on behalf of the merchant
+    ///   - returnURLScheme:  The return URL scheme, e.g. "com.yourcompany.Your-App.payments"
+    ///   - bundleDisplayName: The bundle display name for the current app
+    ///   - environment: The environment, e.g. "production" or "sandbox"
+    ///   - paymentContextID: The Venmo payment context ID (optional)
+    ///   - metadata: Additional Braintree metadata
+    ///   - Returns: The resulting URL, or `nil` if any of the required parameters are `nil`.
+    // TODO: method can be internal once rest of Venmo is in Swift
+    // TODO: consider making these required and doing this check at the call site when BTVenmo_Client is in Swift
+    @objc(appSwitchURLForMerchantID:accessToken:returnURLScheme:bundleDisplayName:environment:paymentContextID:metadata:)
+    public static func appSwitch(
+        forMerchantID merchantID: String?,
+        accessToken: String?,
+        returnURLScheme: String?,
+        bundleDisplayName: String?,
+        environment: String?,
+        paymentContextID: String?,
+        metadata: BTClientMetadata?
+    ) -> URL? {
+        let successReturnURL = returnURL(with: returnURLScheme, result: "success")
+        let errorReturnURL = returnURL(with: returnURLScheme, result: "error")
+        let cancelReturnURL = returnURL(with: returnURLScheme, result: "cancel")
+
+        guard let successReturnURL,
+                let errorReturnURL,
+                let cancelReturnURL,
+                let accessToken,
+                let metadata,
+                let bundleDisplayName,
+                let environment,
+                let merchantID
+        else {
+            return nil
+        }
```

Comment:
```suggestion
        guard [
          successReturnURL,
          errorReturnURL,
          cancelReturnURL,
          accessToken,
          metadata,
          bundleDisplayName,
          environment,
          merchantID
        ].allSatisfy({ $0 != nil }) else {
            return nil
        }
```
Take or leave. It is cool that `guard let` offers the same functionality. This alternate method would prevent having to create shadow variables. I've never used `allSatisfy` before either seems kinda cool.

Reply:
We use all of these later on in `appSwitchParameters` which is why I guard let-ed them - but `allSatisfy` is dope and I am sure we can find a use for it at some point!